File: src/main/java/io/github/kingg22/api_vacunas_panama/persistence/repository/DoctorRepository.java
package io.github.kingg22.api_vacunas_panama.persistence.repository;

import io.github.kingg22.api_vacunas_panama.persistence.entity.Doctor;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface DoctorRepository extends JpaRepository<Doctor, UUID> {

    Optional<Doctor> findByIdoneidad(String idoneidad);

    Optional<Doctor> findByUsuario_Id(UUID idUser);

}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/doctor/repository/DoctorRepository.kt
package io.github.kingg22.api.vacunas.panama.modules.doctor.repository

import io.github.kingg22.api.vacunas.panama.modules.doctor.entity.Doctor
import io.github.kingg22.api.vacunas.panama.util.withSession
import io.quarkus.hibernate.reactive.panache.kotlin.PanacheRepositoryBase
import io.smallrye.mutiny.coroutines.awaitSuspending
import jakarta.enterprise.context.ApplicationScoped
import java.util.UUID

@ApplicationScoped
class DoctorRepository : PanacheRepositoryBase<Doctor, UUID> {
    suspend fun findByIdOrNull(id: UUID): Doctor? = withSession { findById(id).awaitSuspending() }
}
